ec/ecresource/deploymentresource/elasticsearch/v2/elasticsearch_topology.go

```diff
@@ -38,30 +38,32 @@ import (
 )
 
 type ElasticsearchTopologyTF struct {
-	InstanceConfigurationId types.String `tfsdk:"instance_configuration_id"`
-	Size                    types.String `tfsdk:"size"`
-	SizeResource            types.String `tfsdk:"size_resource"`
-	ZoneCount               types.Int64  `tfsdk:"zone_count"`
-	NodeTypeData            types.String `tfsdk:"node_type_data"`
-	NodeTypeMaster          types.String `tfsdk:"node_type_master"`
-	NodeTypeIngest          types.String `tfsdk:"node_type_ingest"`
-	NodeTypeMl              types.String `tfsdk:"node_type_ml"`
-	NodeRoles               types.Set    `tfsdk:"node_roles"`
-	Autoscaling             types.Object `tfsdk:"autoscaling"`
+	InstanceConfigurationId      types.String `tfsdk:"instance_configuration_id"`
+	InstanceConfigurationVersion types.Int64  `tfsdk:"instance_configuration_version"`
+	Size                         types.String `tfsdk:"size"`
+	SizeResource                 types.String `tfsdk:"size_resource"`
+	ZoneCount                    types.Int64  `tfsdk:"zone_count"`
+	NodeTypeData                 types.String `tfsdk:"node_type_data"`
+	NodeTypeMaster               types.String `tfsdk:"node_type_master"`
+	NodeTypeIngest               types.String `tfsdk:"node_type_ingest"`
+	NodeTypeMl                   types.String `tfsdk:"node_type_ml"`
+	NodeRoles                    types.Set    `tfsdk:"node_roles"`
+	Autoscaling                  types.Object `tfsdk:"autoscaling"`
 }
 
 type ElasticsearchTopology struct {
-	id                      string
-	InstanceConfigurationId *string                           `tfsdk:"instance_configuration_id"`
-	Size                    *string                           `tfsdk:"size"`
-	SizeResource            *string                           `tfsdk:"size_resource"`
-	ZoneCount               int                               `tfsdk:"zone_count"`
-	NodeTypeData            *string                           `tfsdk:"node_type_data"`
-	NodeTypeMaster          *string                           `tfsdk:"node_type_master"`
-	NodeTypeIngest          *string                           `tfsdk:"node_type_ingest"`
-	NodeTypeMl              *string                           `tfsdk:"node_type_ml"`
-	NodeRoles               []string                          `tfsdk:"node_roles"`
-	Autoscaling             *ElasticsearchTopologyAutoscaling `tfsdk:"autoscaling"`
+	id                           string
+	InstanceConfigurationId      *string                           `tfsdk:"instance_configuration_id"`
+	InstanceConfigurationVersion int                               `tfsdk:"instance_configuration_version"`
```

Comment:
Does it make sense to use `*int` instead if the attribute's value can be omitted and we'd like to differentiate between `0` and the absence of any value?

Reply:
Actually, I think `int` should work - the backend's model uses `int32` and the field is used only for reading responses from backend.

Reply:
I updated this to `*int` to differentiate between 0 and the absence of any value.